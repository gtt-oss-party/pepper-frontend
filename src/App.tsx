import './App.css';
import { submitWav, isWav } from './Wav';

const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
  const files = event.currentTarget.files;
  if (!files || files?.length === 0) return;
  const file = files[0];
  console.log(file);

  if (isWav(file.name)) {
    submitWav(file.name);
  } else {
    alert('It is not wav');
  }
};

function App() {
  return (
    <>
      <h2>ペッパーくんに物申す</h2>
      <div>
        <input type="file" accept=".wav" aria-label="uploadfile" onChange={handleOnChange} />
      </div>
    </>
  );
}

export default App;
