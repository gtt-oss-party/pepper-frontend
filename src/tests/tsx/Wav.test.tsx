// import { submitWav } from '../../Wav';

import '@testing-library/jest-dom';
import App from '../../App';
import { render, fireEvent } from '@testing-library/react';

describe('send wav file', () => {
  it('should send a text', async () => {
    const { getByLabelText } = render(<App />);
    const file = new File(['wav file content'], 'test.wav', { type: 'audio/wav' });

    (fetch as jest.Mock) = jest.fn().mockResolvedValue('success speech to text');

    const inputElement = getByLabelText('uploadfile') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { files: [file] } });

    expect(console.log).toMatchSnapshot();
  });
  it('should return error', async () => {
    const { getByLabelText } = render(<App />);
    const file = new File(['wav file content'], 'test.wav', { type: 'audio/wav' });

    (fetch as jest.Mock) = jest.fn().mockRejectedValue('error');

    const inputElement = getByLabelText('uploadfile') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { files: [file] } });

    expect(console.error).toMatchSnapshot();
  });
});
