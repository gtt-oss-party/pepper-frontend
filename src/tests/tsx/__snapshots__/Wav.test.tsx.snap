// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`send text file should return error 1`] = `[Function]`;

exports[`send wav file should return error 1`] = `[Function]`;

exports[`send wav file should send a text 1`] = `[Function]`;

exports[`send wav file success 1`] = `[Function]`;

exports[`send wav file success 2`] = `[Function]`;
